pkg/credentialprovider/config.go

```diff
@@ -64,6 +74,32 @@ func GetPreferredDockercfgPath() string {
 }
 
 func ReadDockerConfigFile() (cfg DockerConfig, err error) {
+	// Try happy path first - latest config file
```

Comment:
I think having two separate providers for .docker/config.json" and ".dockercfg" makes sense. It clearly states in the issue that the format is also changed between the two so we cannot use the same secret for both providers.